bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2041,6 +2041,12 @@ private abstract class AbstractImageProviderWrapper {
 	abstract ImageHandle getImageMetadata(int zoom);
 	abstract AbstractImageProviderWrapper createCopy(Image image);
 
+	protected void checkProvider(Object provider, Class<?> expectedClass) {
+		if (provider == null || !expectedClass.isAssignableFrom(provider.getClass())) {
+			SWT.error(SWT.ERROR_NULL_ARGUMENT);
```

Comment:
This is a very bad error for the second case as ist is obviously NOT null but of a wrong type, so it should be `SWT.ERROR_INVALID_ARGUMENT`

Reply:
Keen eye, thank you! #1690